okhttp-tests/src/test/java/okhttp3/internal/ws/RealWebSocketTest.java

```diff
@@ -175,6 +176,8 @@
     server.listener.assertClosing(1000, "Hi!");
     client.listener.assertClosed(1000, "Hello!");
     server.listener.assertClosed(1000, "Hi!");
+
+    client.closeCalled.await(10, TimeUnit.SECONDS);
     assertTrue(client.closed);
```

Comment:
I was getting an intermittent failure here. I believe the reason is that there is a race setting closed to true.